stackrox/stackrox
# chore(ui): Rewrite IntegrationHealthService in TypeScript

## Description

Improve type safety, especially because:
* all imports are in TypeScript files
* corresponding storage types are already in TypeScript

By the way, there are 2 patterns:
* Older API, like this: src/services corresponds to proto/api/v1 and src/types corresponds to proto/storate
* Newer API since early 2022: src/services corresponds to proto/api/v1 and which is **public interface** for response types. On the other side of the coin, proto/storage is **private implementation** even if initially its type have the same properties.

Replace inconsistent `fetchPluginIntegrationsHealth` with `fetchNotifierIntegrationsHealth` function name.

### Learnings

1. Given `get<{ integrationHealth: IntegrationHealth[] }>` TypeScript infers `Promise<IntegrationHealth[]>` return type.

## Checklist

## Testing Performed

1. `yarn lint` in ui
2. `yarn build` in ui
3. `yarn start` in ui

### Manual testing

1. Visit /main/system-health
    * See requests and responses in Network panel.
    * See no errors
        ![system-health_integrationhealth](https://github.com/stackrox/stackrox/assets/11862657/4b5a5b34-441a-451a-95f3-e5d9b9e5bbe3)

### Integration testing

1. `yarn cypress-open` in ui/apps/platform
    * declarativeConfiguration.test.js
    * integrations.test.js
    * vulnDefinitions.test.js
